src/commands/dataset/upload.rs

```diff
@@ -235,6 +217,28 @@ pub async fn upload(args: Upload, global: GlobalArgs) -> Result<()> {
         }
     };
 
+    let write_options = args.write.parse()?;
+
+    let (schema, stream) = open(
```

Comment:
I put the open function before the mutation here because I wanted to avoid doing a mutation if I can't even open the file, but I understand that its probably nicer UX this way? so its up to you